### PR TITLE
LibC: Implemented `strcasestr()`

### LINKED ISSUE #13214: LibC: Implement `strcasestr()`, `strcasecmp()`, `strncasecmp()` in `string.h`
I was messing around with porting some software when I discovered a few functions that were missing from `string.h` library in LibC:

`strcasestr()`: Does exactly what `strstr()` does, except it ignores case. ([manpage](https://www.man7.org/linux/man-pages/man3/strstr.3.html))
`strcasecmp()`: Does exactly what `strcmp()` does, but it ignores case ([manpage](https://man7.org/linux/man-pages/man3/strcasecmp.3.html))
`strncasecmp()`: Does exactly what `strncmp()` does, but it ignores case ([manpage](https://man7.org/linux/man-pages/man3/strcasecmp.3.html))

Note that these two functions are part of POSIX.

I tried to implement them myself, but I'm stupid and caused a kernel panic `:^(`

### PR DESCRIPTION
**Pull request by stautonico**
This function is almost the same as `strstr()` except for the fact that it ignores case in `haystack` and the `needle`.

This function is not part of POSIX, however, it is included in Linux and BSD and some programs (htop for example) does rely on it.

Fixes #13214

### CODE REVIEW

**Comment by mustafaquraish** [comment]
Hey, welcome to Serenity!

We don't do fixup-commits in the same PR, so please merge the second fixing the formatting with the first one. If you're unsure how, please join to discord and ask one of us (or @ me), and we'll help you out!

File: Userland/Libraries/LibC/string.cpp [hunk 1]
```diff
@@ -368,6 +368,23 @@ char* strstr(const char* haystack, const char* needle)
...
+    char nch;
+    char hch;
+
+    if ((nch = *needle++) != 0) {
+        size_t len = strlen(needle);
+        do {
+            do {
+                if ((hch = *haystack++) == 0)
+                    return nullptr;
+            } while (hch != nch);
```
Comment (timschumi) [inline Userland/Libraries/LibC/string.cpp:382]: Won't this fast-forward beyond a potential starting point of the needle, as it is case-sensitive?

**Comment by sin-ack** [comment]
Hi there. We don't really add "dead code" (code with no users) to the system unless something like a port relies on it. Does a port rely on this function existing, or is it otherwise used?

**Comment by archanox** [comment]
> Hi there. We don't really add "dead code" (code with no users) to the system unless something like a port relies on it. Does a port rely on this function existing, or is it otherwise used?

In the original comment for this PR it says that this is a requirement for getting `htop` ported to SerenityOS. So I guess it is considered dead code until the port is ported, but it is something that I'd personally like to see working on Serenity.

**Comment by mustafaquraish** [comment]
> In the original comment for this PR it says that this is a requirement for getting `htop` ported to SerenityOS. So I guess it is considered dead code until the port is ported, but it is something that I'd personally like to see working on Serenity.

If it's for a port, ideally this change should be a commit should be part of the PR that also includes the port of `htop`.

**Comment by stautonico** [comment]
I didn't add this strictly for htop, I just came across it while trying to port htop. I couldn't get the htop port working (yet), so I guess for now it's considered dead code

File: Userland/Libraries/LibC/string.cpp (hunk 1 above)
Comment (stautonico) [reply Userland/Libraries/LibC/string.cpp:382]: I don't think so, the `case` class of functions are not case sensitive while the standard ones are.

Comment (timschumi) [reply Userland/Libraries/LibC/string.cpp:382]: Sure, the functions are not, but raw comparisons of `char` are. More precisely, you are missing `tolower` or something similar around those character comparisons here.

**Comment by sin-ack** [comment]
That's fine, you can set this PR as a draft in the meantime and then add your htop port commit on top of it afterwards.

**Comment by bgianfo** [comment]
I don't see any reason to require the port to be finished before extending LibC with this functionality. As long as there are unit tests that cover it. While not standardized by Dr. POSIX, this is a completely normal API for any modern Unix LibC to support.

**Comment by stautonico** [comment]
As a noob to this project, I'm not going to argue either side, but I can definitely understand why this shouldn't be added since no one uses it (yet), but it is a common function used in both Linux and BSD (even though its not POSIX).

**Pull request closed without merging**